toolkit/tools/internal/pkgjson/pkgjson_test.go

```diff
@@ -257,25 +284,116 @@ func TestSubsetValidOutside(t *testing.T) {
 
 	assert.True(t, interval1.Satisfies(&interval2))
 	assert.True(t, interval1.Satisfies(&interval3))
-	assert.False(t, interval1.Contains(&interval2))
-	assert.False(t, interval1.Contains(&interval3))
+	assert.True(t, interval1.GuaranteedSatisfies(&interval2))
+	assert.True(t, interval1.GuaranteedSatisfies(&interval3))
 }
 
 func TestSubsetInvalid(t *testing.T) {
 	p1 := &PackageVer{Version: "1", Condition: ">=", SVersion: "3", SCondition: "<="}
 	p2 := &PackageVer{Version: "1", Condition: ">"}
-	p3 := &PackageVer{Version: "1", Condition: ">"}
 	interval1, err := p1.Interval()
 	assert.NoError(t, err)
 	interval2, err := p2.Interval()
 	assert.NoError(t, err)
-	interval3, err := p3.Interval()
-	assert.NoError(t, err)
 
 	assert.True(t, interval1.Satisfies(&interval2))
-	assert.True(t, interval1.Satisfies(&interval3))
-	assert.False(t, interval1.Contains(&interval2))
-	assert.False(t, interval1.Contains(&interval3))
+	assert.False(t, interval1.GuaranteedSatisfies(&interval2))
+}
+
+// Ensure we don't allow 'pkg > 1' and 'pkg < 3' to match. While they have overlapping versions, in practice we have no
+// guarantee that we will actually be able to provide a version that satisfies both conditions.
+func TestSubsetNonDeterministicInterval(t *testing.T) {
+	testInputs := []struct {
+		name                                                                 string
+		existingPkgVer1, existingPkgCond1, existingPkgVer2, existingPkgCond2 string
+		queryPkgVer1, queryPkgCond1, queryPkgVer2, queryPkgCond2             string
+		expected                                                             bool
+	}{
+		{
+			name:            "p < q, unbounded",
+			existingPkgVer1: "1", existingPkgCond1: ">",
+			queryPkgVer1: "3", queryPkgCond1: ">",
```

Comment:
Why not use `PackageVer` immediately, since we use the strings only to create `PackageVer` objects anyway:
```suggestion
		existingPkg PackageVer
		queryPkg PackageVer
		expected                                                             bool
	}{
		{
			name:            "p < q, unbounded",
			existingPkgVer1: PackageVer{Version: "1", Condition: "<"},
			queryPkgVer1: PackageVer{Version: "2", Condition: ">"},
```